todo/models.py

```diff
@@ -73,14 +124,35 @@ class Comment(models.Model):
     a comment and change task details at the same time. Rolling our own since it's easy.
     """
 
-    author = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    author = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, blank=True, null=True
+    )
     task = models.ForeignKey(Task, on_delete=models.CASCADE)
     date = models.DateTimeField(default=datetime.datetime.now)
+    email_from = models.CharField(max_length=320, blank=True, null=True)
+    email_message_id = models.TextField(blank=True, null=True)
+
     body = models.TextField(blank=True)
 
+    class Meta:
+        # an email should only appear once per task
+        unique_together = ("task", "email_message_id")
```

Comment:
This `unique_together` should enforce the "one thread per message id" I referred to above, so we shouldn't need to guess there by counting comments. Unless I am misunderstanding.

Reply:
I don't see how this constraint helps. Each incoming emails can reference many other messages. What if most of these references are from one task, and one is from another task ? If I had to choose, I would attach the incoming message to the task having most related comments. That what the current code does.

Reply:
Yep, I get it now.